pkg/version/info.go

```diff
@@ -0,0 +1,58 @@
+package version
+
+import (
+	"bytes"
+	"runtime"
+	"strings"
+	"text/template"
+)
+
+// These are mostly set during compilation from linker flags
+var (
+	application = "tuwat"
+	version     = "dev"
+	revision    string
+	branch      string
+	buildDate   string
+)
+
+// A rich representation of the version information which can also be readily serialized into a JSON representation
+type VersionInfo struct {
+	Application string `json:"application"`
+	Version     string `json:"version"`
+	Revision    string `json:"revision,omitempty"`
+	Branch      string `json:"branch,omitempty"`
+	BuildDate   string `json:"buildDate,omitempty"`
+	GoVersion   string `json:"goVersion"`
+	GoPlatform  string `json:"goPlatform"`
+}
+
+var versionInfoTmpl = `
+{{.Application}}, version {{.Version}} (branch: {{.Branch}}, revision: {{.Revision}})
+  build date:       {{.BuildDate}}
+  go version:       {{.GoVersion}}
+  platform:         {{.GoPlatform}}
+`
+
+func (v VersionInfo) Print() string {
```

Comment:
Not sure about the name.  `Print()` suggests that it prints something, but it only returns a string.

Reply:
It is now called `.HumanReadable()` (see 47df78f)